File: demos/go/test-go/utils/kv/json_engine.go
package kv

import (
	"encoding/json"
	"os"
)

type JsonEngine struct {
	filename string
	config   map[string]interface{}
}

func NewJsonEngine(filename string) *JsonEngine {
	config, _ := readJsonFile(filename)

	return &JsonEngine{
		filename: filename,
		config:   config,
	}
}

func WriteJsonFile(filename string, config map[string]interface{}) error {
	contents, _ := json.Marshal(config)
	return os.WriteFile(filename, []byte(contents), 0644)
}

func readJsonFile(filename string) (map[string]interface{}, error) {
	config := make(map[string]interface{})
	data, err := os.ReadFile(filename)
	if err == nil {
		json.Unmarshal(data, &config)
	}
	return config, nil
}

func (j *JsonEngine) Save() error {
	return WriteJsonFile(j.filename, j.config)
}

func (j *JsonEngine) set(key string, val interface{}) {
	j.config[key] = val
}

func (j *JsonEngine) get(key string) interface{} {
	return j.config[key]
}

func (j *JsonEngine) remove(key string) {
	delete(j.config, key)
}

func (j *JsonEngine) keys() []string {
	var keys []string
	for key := range j.config {
		keys = append(keys, key)
	}
	return keys
}

File: demos/py/scraping/requests-demo/kv/json_engine.py
import json
import os
from typing import Any, Dict, List, Optional
from . import KvEngine

class JsonEngine(KvEngine):
    def __init__(self, filename: str):
        self.filename = filename
        self.config = self.read_json_file(filename)

    def read_json_file(self, filename: str) -> Dict[str, Any]:
        try:
            with open(filename, 'r') as file:
                return json.load(file)
        except (FileNotFoundError, json.JSONDecodeError):
            return {}

    def write_json_file(self, filename: str, config: Dict[str, Any]) -> None:
        with open(filename, 'w') as file:
            json.dump(config, file, indent=4)

    def save(self) -> None:
        self.write_json_file(self.filename, self.config)

    def set(self, key: str, val: Any) -> None:
        self.config[key] = val

    def get(self, key: str) -> Optional[Any]:
        return self.config.get(key)

    def remove(self, key: str) -> None:
        self.config.pop(key, None)

    def keys(self) -> List[str]:
        return list(self.config.keys())